debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -143,35 +209,61 @@ public static long discontiguousHeaderSize()
 	}
 
 	/**
-	 * @param objPointer the contiguous array object who's dataAddr field we are accessing
-	 * @throws CorruptDataException If there's a problem accessing the indexable object dataAddr field
-	 * @throws NoSuchFieldException If the indexable object dataAddr field does not exist on the build that generated the core file
+	 * Get the header size of the discontiguous indexable object with the dataAddress header field.
+	 */
+	public static long discontiguousHeaderSizeWithDataAddress()
+	{
+		if (mixedReferenceMode) {
+			if (compressObjectReferences) {
+				return J9IndexableObjectWithDataAddressDiscontiguousCompressed.SIZEOF;
+			}
+			return J9IndexableObjectWithDataAddressDiscontiguousFull.SIZEOF;
+		}
+		return J9IndexableObjectWithDataAddressDiscontiguous.SIZEOF;
+	}
+
+	/**
+	 * @param objPointer the contiguous array object who's dataAddr field we are accessing.
+	 * @throws CorruptDataException If there's a problem accessing the indexable object dataAddr field.
+	 * @throws NoSuchFieldException If the indexable object dataAddr field does not exist on the build that generated the core file.
 	 */
 	public static VoidPointer getDataAddrForContiguous(J9IndexableObjectPointer objPointer) throws CorruptDataException, NoSuchFieldException
 	{
 		if (mixedReferenceMode) {
 			if (compressObjectReferences) {
-				return VoidPointer.cast((J9IndexableObjectContiguousCompressedPointer.cast(objPointer)).dataAddr());
+				return VoidPointer.cast((J9IndexableObjectWithDataAddressContiguousCompressedPointer.cast(objPointer)).dataAddr());
 			}
-			return VoidPointer.cast((J9IndexableObjectContiguousFullPointer.cast(objPointer)).dataAddr());
+			return VoidPointer.cast((J9IndexableObjectWithDataAddressContiguousFullPointer.cast(objPointer)).dataAddr());
 		}
-		return VoidPointer.cast((J9IndexableObjectContiguousPointer.cast(objPointer)).dataAddr());
+		return VoidPointer.cast((J9IndexableObjectWithDataAddressContiguousPointer.cast(objPointer)).dataAddr());
```

Comment:
While you're here, please remove the redundant parentheses (also in `getDataAddrForDiscontiguous()`).